tests/testthat/test-predictions.R

```diff
@@ -0,0 +1,38 @@
+context("predictions")
+
+test_that("prediction for gaussian models peform as expected", {
+  set.seed(1)
+
+  x <- with(trees, cbind(Girth, Volume))
+  y <- trees$Height
+
+  fit <- sgdnet(x, y)
+
+  pred_link <- predict(fit, x, type = "link")
+  pred_response <- predict(fit, x, type = "response")
+
+  # link and response should be the same for the gaussian model
+  expect_equal(pred_link, pred_response)
+
+  pred_nonzero <- predict(fit, x, type = "nonzero")
+  expect_is(pred_nonzero, "list")
+
+  pred_coefficients <- predict(fit, x, type = "coefficients")
+  expect_equal(pred_coefficients, coef(fit))
+
+  # check linear interpolation
+  pred_new <- predict(fit, x, s = 0.04)
+  expect_is(pred_new, "matrix")
```

Comment:
Can we check that these tests are statistically right in addition to returning the right type of objects?

Reply:
Could you please specify what you are after? I've added a test for the linear predictors but checking the weights seems just like repeating what we are already doing in [test-gaussian.R](https://github.com/jolars/sgdnet/blob/master/tests/testthat/test-gaussian.R), no?

Reply:
For example, here you test that the interpolation returns a matrix, but you don't actually check that it's statistically correct. You should also have a test along the lines of  `expect_equal(pred_new, 0.5 * predict(fit, x, s=0.03) + 0.5 * predict(fit, x, s=0.05))` [that's not actually correct, but you get the idea.] I don't think this is in the Gaussian tests yet, but I might be missing it. 

Similarly, to check pred_nonzero and pred_coefficients, I'd at least check something like `expect_equal(pred_nonzero, pred_coefficients != 0)`.